app/src/components/submissionPage/TextBoxIfSnapcupOpen.tsx

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import MentionElements from "../../types/MentionElements";
+import SubmissionTextBox from "./SubmissionTextBox";
+import { getCurrentUserName } from "../../firebase/users/UserService";
+import SubmissionBoxWrapper from "./SubmissionBoxWrapper";
+
+const MessageInBox = ({ message }) => (
+    <SubmissionBoxWrapper>
+        <div className="col text-light">
+            <h3>{message}</h3>
+        </div>
+    </SubmissionBoxWrapper>
+);
+
+const TextBoxIfSnapcupOpen = ({ cup, snappables, status }) => {
+    if (cup?.isOpen) {
+        return (
+            <SubmissionTextBox
+                cup={cup}
+                snappables={snappables}
+                user={getCurrentUserName()}
+            />
+        );
+    } else if (cup) {
+        return (
+            <MessageInBox message="Apologies, the SnapCup is closed for new submissions." />
+        );
+    } else if (status) {
+        return <MessageInBox message={status} />;
+    } else {
+        return (
+            <MessageInBox message="Apologies, there is no open SnapCup at the moment." />
+        );
+    }
+};
+
```

Comment:
I think some refactoring is in order.

1. I think the `SubmissionBoxWrapper` should be referenced once by this component, and not included separately in  `SubmissionTextBox`.
2. The logic of deciding what is the main content of this component should be a dedicated method.
3. There should only be a single return statement. (e.g:

```javascript
return (
<SubmissionBoxWrapper>
  {getContent()}
</SubmissionBoxWrapper>
)
```

Reply:
Actually, I'm going to suggest that the `SubmissionBoxWrapper` gets merged into this component entirely.